newIDE/app/src/MainFrame/index.js

```diff
@@ -856,25 +913,34 @@ class MainFrame extends React.Component<Props, State> {
         editorTabs: openEditorTab(this.state.editorTabs, {
           name,
           renderEditor: ({ isActive, editorRef }) => (
-            <ExternalLayoutEditor
-              project={this.state.currentProject}
-              externalLayoutName={name}
-              setToolbar={this.setEditorToolbar}
-              onPreview={this._launchExternalLayoutPreview}
-              showPreviewButton={!!this.props.previewLauncher}
-              showNetworkPreviewButton={
-                this._previewLauncher &&
-                this._previewLauncher.canDoNetworkPreview()
-              }
-              onOpenDebugger={this.openDebugger}
-              onEditObject={this.props.onEditObject}
-              showObjectsList={!this.props.integratedEditor}
-              resourceSources={this.props.resourceSources}
-              onChooseResource={this._onChooseResource}
-              resourceExternalEditors={this.props.resourceExternalEditors}
-              isActive={isActive}
-              ref={editorRef}
-            />
+            <PreferencesContext.Consumer>
```

Comment:
Not a big fan of having to use PreferencesContext.Consumer here (it's verbose)... but I don't have a better solution for now.

Reply:
Yeah me too. Was wondering if I can get the value out without rendering the prop